### ISSUE TITLE
Action "Send email" - wrong URL when using behind http reverse proxy

### ISSUE THREAD

**Issue by cljk** [oroinc/crm#285]
I installed OroCRM 2.3.2 on a server with a private IP (192.168.10.231) and made it public visible with a http reverse proxy based on Apache 2.x - lets call it "crm.company.com".

The system configuration "application URL" is set to the external name "https://crm.company.com".

The Apache config for this is like

```
 ProxyPass /  http://192.168.10.231/
 ProxyPassReverse /   http://192.168.10.231

```

So far everything is working as expected... but there is one issue:
When I want to send an email for example to an account the HTML form action is set to the internal IP (192.168.10.231). The submit then fails because the internal IP is not accessible from external. 

`<form id="oro_email_email-uid-59a6b21f3ce21" name="oro_email_email" method="post" action="http://192.168.10.231/email/create?_wid=ebb7053d-7b49-4c34-93e2-2c5042780e0b&amp;_widgetContainer=dialog&amp;entityClass=Oro_Bundle_AccountBundle_Entity_Account&amp;entityId=2&amp;to=" enctype="multipart/form-data" data-bound-view="oroform/js/app/views/form-validate-view" novalidate="novalidate">`

The other actions like "call" or "task" generate relative paths which works.
The email-popup is rendered by an ajax-call to URI
`https://crm.company.com/email/create?to=marcel.pokrandt%40vgb.org&entityClass=Oro_Bundle_UserBundle_Entity_User&entityId=3&_widgetContainer=dialog&_wid=1d44ec6f-6e28-43b9-82a1-fd2277d1581e`

![image](https://user-images.githubusercontent.com/654602/29873017-29d8b1a2-8d92-11e7-81ab-11e6768be28e.png)

Is there a way to fix this URL-issue? Why is the system using absolute paths and even includes the server IP and not the application url?

**Comment by manowark** [comment]
Hi, @cljk.

Could you please provide the result of SQL query on your database?
```
SELECT cv.name, cv.text_value
FROM oro_config_value AS cv
WHERE cv.name LIKE '%url';
```

**Comment by cljk** [comment]
sure... checked that before ;-)

```
mysql> SELECT cv.name, cv.text_value
    -> FROM oro_config_value AS cv
    -> WHERE cv.name LIKE '%url';
+-----------------+---------------------+
| name            | text_value          |
+-----------------+---------------------+
| application_url | https://crm.vgb.org |
+-----------------+---------------------+
1 row in set (0,00 sec)
```
(crm.vgb.org what I replaced with crm.company.com in previous post)

**Comment by cljk** [comment]
PS
The cache is clean too ...

**Comment by manowark** [comment]
Ok, I found that. Problem in line https://github.com/oroinc/platform/blob/master/src/Oro/Bundle/EmailBundle/Resources/views/Email/dialog/update.html.twig#L37.

Looks like should be `{{ formAction }}` instead. `formAction` should be passed from controller`s action and it can be generated by next code
```
$formAction = $this->get('router')->generate('oro_email_email_create');
```

**Comment by cljk** [comment]
Sounds promising - I´ll check that tomorrow and give feedback.

**Comment by cljk** [comment]
I tried another solution first - adding following lines to the `app.php` (compare https://symfony.com/blog/fixing-the-trusted-proxies-configuration-for-symfony-3-3)

```
Request::setTrustedHeaderName(Request::HEADER_FORWARDED, null);
Request::setTrustedProxies(['192.168.10.0/24']);
```
This kind of "works" but generates the URL with "http" instead of "https" - and gets blocked by the browser because of "mixed content".

```
<form id="oro_email_email-uid-59a7a8cf18c71" name="oro_email_email" method="post" action="http://crm.vgb.org/email/create?_wid=f6e22b6a-dcbc-4979-b4da-307e280275cb&amp;_widgetContainer=dialog&amp;entityClass=Oro_Bundle_AccountBundle_Entity_Account&amp;entityId=2&amp;to=" enctype="multipart/form-data" data-bound-view="oroform/js/app/views/form-validate-view" novalidate="novalidate">
```

I´ll check out your idea later. 
If this won´t work either I´ll change the communication sheme proxy -> webserver from HTTP to HTTPS. I think this at least will be a valid workaround.

**Comment by cljk** [comment]
Your idea was a good one ... thank you for the hint .... but could not solve it all. 

I checked out the "platform" sources and searched for "app.request.*". There are too much bundles which reference the `app.request.uri` or `app.request.scheme` than I thought. I double checked some of them and found at least a problem editing the menu and generating some javascripts with requireJs.

I simply can´t fix them all. :-(

So my only way to go is to use the modified `app.php` (see above) and switch internal communication schema between proxy and webserver to HTTPS. 
I did that and all seems to work now.... 

I´ll file an issue at the platform project and close this issue here.
